src/libPMacc/include/memory/buffers/HostBufferIntern.hpp

```diff
@@ -101,16 +102,36 @@ class HostBufferIntern : public HostBuffer<TYPE, DIM>
         __startOperation(ITask::TASK_HOST);
         this->setCurrentSize(this->getDataSpace().productOfComponents());
         if (!preserveData)
-            memset(pointer, 0, this->getDataSpace().productOfComponents() * sizeof (TYPE));
+        {
+            /* if it is a pointer out of other memory we can not assume that
+             * that the physical memory is contiguous 
+             */
+            if(ownPointer)
+                memset(pointer, 0, this->getDataSpace().productOfComponents() * sizeof (TYPE));
+            else
+            {
+                TYPE value;
+                uint8_t* valuePtr = (uint8_t*)&value;
+                for( size_t b = 0; b < sizeof(TYPE); ++b)
+                {
+                    valuePtr[b] = static_cast<uint8_t>(0);
+                }
+                setValue(value);
+            }
+        }
     }
 
     void setValue(const TYPE& value)
     {
         __startOperation(ITask::TASK_HOST);
         size_t current_size = this->getCurrentSize();
+        PMACC_AUTO(memBox,getDataBox());
+        typedef DataBoxDim1Access<DataBoxType > D1Box;
+        D1Box d1Box(memBox, this->getDataSpace());
+        #pragma omp parallel for
         for (size_t i = 0; i < current_size; i++)
```

Comment:
It is only possible to use `std::fill` memory line wise (each line can have a padding at the end). The current method is the easiest way to fill a Buffer.